app_allocator/tests/test_judge.py

```diff
@@ -9,3 +15,53 @@ def test_judge_mark_as_done_creates_event(self):
         judge.mark_as_done()
         assert Event.ticks == prior_ticks + 1
         assert Event.all_events[-1].fields['subject'] == judge
+
+    def test_judge_mark_as_done_zeroes_judge_remaining_count(self):
+        judge = Judge()
+        judge.remaining = 10
+        judge.mark_as_done()
+        assert judge.remaining == 0
+
+    def test_complete_startups_creates_events_for_all_startups(self):
+        judge = Judge()
+        startups = [Startup() for _ in range(10)]
+        judge.startups = startups[:]
+        events = judge.complete_startups()
+        event_startups = [event.fields['object'] for event in events]
+        assert all([startup in event_startups for startup in startups])
+
+    @mock.patch("app_allocator.classes.judge.Judge.passes", always_pass)
```

Comment:
Nice way to work around the random thing.  We should switch to using seeds, but this works.
Not a blocker, but from a logical completeness perspective we should probably have an equivalent for the "finished" case.  It's really a side effect of how the code is implemented that this case doesn't show up in the coverage report and if it weren't true, it would be a problem.

Reply:
Yeah, I'd love to push past complete line coverage to something more like logical or even semantic coverage, but this ticket was mostly about lines.